### ISSUE TITLE
Add PDF reports while using split-merge

### ISSUE THREAD

**Issue by pierotofy** [OpenDroneMap/ODM#1227]
Currently ODM does not generate PDF reports when using split-merge. It could be possible to generate them by:

- Recomputing/aggregating stats from submodels
- Write a new report generation code (separate from OpenSfM's)

**Comment by smathermather** [comment]
I'd lean toward aggregate stats. Any reason to not have submodel stats as well?  For larger datasets it could be more useful than global stats.

**Comment by pierotofy** [comment]
I agree to both. The bottom line is that the split-merge report would contain slightly different information compared to normal runs.

**Comment by smathermather** [comment]
Yup. Agreed.

**Comment by YanNoun** [comment]
@pierotofy `opensfm/action/compute_statistics.py` calls `stats.py` functions that works on a list of reconstructions.

**Comment by pierotofy** [comment]
Didn't know that. Thanks @YanNoun ! I'll try it out.

**Comment by YanNoun** [comment]
The `save_heatmap`/`save_topview` might be super slow with 100K/1M points, feel free to run some random index-based decimation.

**Comment by pierotofy** [comment]
I did notice some slowdowns with larger datasets; would this be a good idea for a PR to OpenSfM for the `compute_statistics` command? (I could add a `--decimate` flag)

**Comment by YanNoun** [comment]
Yup definitely, that would be great !

**Comment by pierotofy** [comment]
https://github.com/mapillary/OpenSfM/pull/695

I opted to allow a user to set a cap on the number of points using random sampling. Definitely helps keep runtime in check!
